src/Jenssegers/Mongodb/Concerns/TransactionManager.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Jenssegers\Mongodb\Concerns;
+
+use Closure;
+use Exception;
+use MongoDB\Driver\Session;
+
+use function MongoDB\with_transaction;
+
+trait TransactionManager
+{
+    /**
+     * create a session and start a transaction in session
+     *
+     * In version 4.0, MongoDB supports multi-document transactions on replica sets.
+     * In version 4.2, MongoDB introduces distributed transactions, which adds support for multi-document transactions on sharded clusters and incorporates the existing support for multi-document transactions on replica sets.
+     * To use transactions on MongoDB 4.2 deployments(replica sets and sharded clusters), clients must use MongoDB drivers updated for MongoDB 4.2.
+     *
+     * @see https://docs.mongodb.com/manual/core/transactions/
+     * @return void
+     */
+    public function beginTransaction(?array $options = [])
```

Comment:
The options parameter doesn't need to be nullable; the important part is having a default value so the method is compatible with the parent interface.

```suggestion
    public function beginTransaction(array $options = [])
```

Reply:
Yes, but startTransaction declaration has nullable argument that is why  I decide to define it as nullable 
`final public function startTransaction(?array $options = []) {}`

Reply:
That is a remnant of the current API and will be changed in 2.0, so I'd prefer to pre-empt that change here